libs/sdk-react-native/ios/RNBreezSDK.swift

```diff
@@ -1,4 +1,42 @@
 import Foundation
+#if canImport(BreezSDK)
+import BreezSDK
+#else
+enum BreezSDK {
```

Comment:
This is a bit odd. It works though and doesn't affect the production release. Only local dev setups. If anyone has a better idea to solve this let me know! 🙏 😄

Reply:
@cnixbtc I think if we import a module in swift we can use the functions without the module prefix so we don't need to maintain this proxy enum. Did you try that?

Reply:
I actually think this is a pretty cool approach 👏

Reply:
> @cnixbtc I think if we import a module in swift we can use the functions without the module prefix so we don't need to maintain this proxy enum. Did you try that?

Would the module function names not conflict with the class function names?

Reply:
> @cnixbtc I think if we import a module in swift we can use the functions without the module prefix so we don't need to maintain this proxy enum. Did you try that?

The module is already explicitly imported. In the locally-build Breez SDK scenario both the SDK and the RN plugin are the same module: `breez_sdk`.

<img width="933" alt="Screenshot 2023-06-22 at 21 35 02" src="https://github.com/breez/breez-sdk/assets/111755602/da63f047-539d-4ad2-bbc5-c175a9f423ea">

---

The problem is pretty much what @dangeross said:

> Would the module function names not conflict with the class function names?

That's correct. We need to disambiguate e.g., `mnemonicFromSeed` in `BreezSDK.swift` from `mnemonicFromSeed` in `RNBreezSDK.swift`:

<img width="862" alt="Screenshot 2023-06-22 at 21 36 49" src="https://github.com/breez/breez-sdk/assets/111755602/c36d53fa-73f6-47e5-96a8-c95f5980ad2e">

---

That's why we have to explicitly prefix the module name to make this clear. The simplest way to do this would be to rename the RN plugin module to `BreezSDK` from `breez_sdk` but we cannot do this since then the module name would clash with the `BreezSDK` CocoaPod in the production scenario.

I hope this makes sense, let me know if not. Took me quite a while to figure it out myself. 😅

Reply:
Understood. Another alternative comes to my mind is to refer to a localy built pod package (which have the same namespace as production) is it possible?

Reply:
Oh, I like that idea! Will try that out tomorrow.

Reply:
An alternative podspec for local development would work as well. The only inconvenient thing I noticed while testing was that in development mode (when the locally built Breez SDK is part of the RN plugin), we'll get this warning in the plugin's Xcode project:

<img width="1328" alt="Screenshot 2023-06-25 at 11 10 51" src="https://github.com/breez/breez-sdk/assets/111755602/d73b41eb-dc55-4418-8307-c3e8209d4caf">

Let me know which approach you guys would prefer. I'm slightly erring towards the alternative podspec since that would drop the need to manually update the namespace proxy in case something changes (which probably wouldn't happen often anyway—but still...).

Reply:
@cnixbtc I am tending towards using local podspec and agree with the reason you mentioned (not having to maintain the proxy).
What are the local changes needed in order to run in dev mode?

Reply:
Nice. I've made the change and also added some developer docs. Basically you do this:

You start out with the prod podspec being active:

- `breez_sdk.podspec`

and the dev podspec being disabled:

- `BreezSDK.podspec.dev`

To switch to dev mode you need to rename the files:

- `breez_sdk.podspec.prod`
- `BreezSDK.podspec`

The dev/prod file endings can be whatever. Whatever ends in `podspec` will be taken by React Native. Imo it would be cool if you could control what podspec react native uses that in a better way (e.g., via a config) but a quick search turned up no results. I think the file-renaming is an acceptable way of doing it for now, though.

Reply:
I agree.

Reply:
Looks good